features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/FlagsClient.kt

```diff
@@ -77,109 +108,351 @@ interface FlagsClient {
      */
     fun resolveStructureValue(flagKey: String, defaultValue: JSONObject): JSONObject
 
+    /**
+     * Builder for creating [FlagsClient] instances with custom configuration.
+     *
+     * The builder uses a selective override pattern: configuration fields set explicitly
+     * on the builder override the defaults from [FlagsFeature]. Fields not set on the
+     * builder use the feature-level defaults.
+     *
+     * ## Usage Examples
+     *
+     * Default [FlagsClient]:
+     * ```
+     * val client = FlagsClient.Builder().build()
+     * ```
+     *
+     * Named [FlagsClient]:
+     * ```
+     * val client = FlagsClient.Builder("analytics").build()
+     * ```
+     *
+     * With custom configuration:
+     * ```
+     * val client = FlagsClient.Builder("analytics")
+     *     .useCustomEndpoint("https://custom.endpoint.com")
+     *     .build()
+     * ```
+     *
+     * With custom SDK core:
+     * ```
+     * val client = FlagsClient.Builder(name = "analytics", sdkCore = customCore)
+     *     .useFlaggingProxy("https://proxy.example.com")
+     *     .build()
+     * ```
+     */
+    class Builder {
+        private val name: String
+        private val sdkCore: SdkCore
+
+        // Optional configuration overrides (null = use feature default)
+        private var explicitCustomEndpoint: String? = null
+        private var explicitFlaggingProxy: String? = null
+        private var explicitEnableExposureLogging: Boolean? = null
```

Comment:
I think this is where confusion may be coming: we have also similar params in `FlagsConfiguration`. I think we should have a single place where we set them up.